pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -425,3 +383,74 @@ func proxyMapToEnvVars(proxyConfig map[string]string) []corev1.EnvVar {
 	sort.Slice(envVars, func(i, j int) bool { return envVars[i].Name < envVars[j].Name })
 	return envVars
 }
+
+func gracefulTerminationDurationFromConfig(operatorSpec *operatorv1.StaticPodOperatorSpec) (int, error) {
+	var watchTerminationDurationPath = []string{"watchTerminationDuration"}
+
+	mergedConfigs, err := resourcemerge.MergeProcessConfig(map[string]resourcemerge.MergeFunc{}, operatorSpec.ObservedConfig.Raw, operatorSpec.UnsupportedConfigOverrides.Raw)
+	if err != nil {
+		return 0, err
+	}
+
+	// read the watch termination from the observed configuration
+	observedConfig := map[string]interface{}{}
+	if err := json.NewDecoder(bytes.NewBuffer(mergedConfigs)).Decode(&observedConfig); err != nil {
+		return 0, err
+	}
+	observedWatchTerminationDurationStr, _, err := unstructured.NestedString(observedConfig, watchTerminationDurationPath...)
+	if err != nil {
+		return 0, fmt.Errorf("unable to extract watchTerminationDuration from the observed config: %v, path = %v", err, watchTerminationDurationPath)
+	}
+	observedWatchTerminationDuration, err := strconv.Atoi(observedWatchTerminationDurationStr)
+	if err != nil {
+		return 0, fmt.Errorf("incorrect value of watchTerminationDuration field in the observed config: %v", err)
+	}
+
+	return observedWatchTerminationDuration, nil
+}
+
+type kasTemplate struct {
+	Image                       string
+	OperatorImage               string
+	Verbosity                   string
+	GracefulTerminationDuration int
+}
+
+func manageTemplate(rawTemplate string, imagePullSpec string, operatorImagePullSpec string, operatorSpec *operatorv1.StaticPodOperatorSpec) (string, error) {
+	var verbosity string
+	switch operatorSpec.LogLevel {
+	case operatorv1.Normal:
+		verbosity = fmt.Sprintf(" -v=%d", 2)
+	case operatorv1.Debug:
+		verbosity = fmt.Sprintf(" -v=%d", 4)
+	case operatorv1.Trace:
+		verbosity = fmt.Sprintf(" -v=%d", 6)
+	case operatorv1.TraceAll:
+		verbosity = fmt.Sprintf(" -v=%d", 8)
+	default:
+		verbosity = fmt.Sprintf(" -v=%d", 2)
+	}
+
+	terminationDuration, err := gracefulTerminationDurationFromConfig(operatorSpec)
```

Comment:
why don't you pick the graceful termination duration directly from the server flags? Why using another field in the config?